src/stores/index.js

```diff
@@ -1011,28 +1077,32 @@ const Store = Reflux.createStore({
     this.state.hasUnsavedChanges = false;
     this.state.isURIEditable = false;
 
-    if (this.state.viewType === 'connectionString') {
-      Connection.from(url, (error, parsedConnection) => {
-        if (!error) {
-          currentConnection.set(this._getPoorAttributes(parsedConnection));
+    if (this.state.viewType === CONNECT_VIEWS.CONNECTION_STRING) {
+      try {
+        const buildConnectionModelFromUrl = promisify(Connection.from);
+        const parsedConnection = await buildConnectionModelFromUrl(url);
 
-          if (url.match(/[?&]ssl=true/i)) {
-            currentConnection.sslMethod = 'SYSTEMCA';
-          }
+        currentConnection.set(this._getPoorAttributes(parsedConnection));
 
-          this._saveConnection(currentConnection);
+        if (url.match(/[?&]ssl=true/i)) {
+          currentConnection.sslMethod = 'SYSTEMCA';
         }
-      });
-    } else if (!currentConnection.isValid()) {
-      const validationError = currentConnection.validate(currentConnection);
 
-      this.setState({
-        isValid: false,
-        errorMessage: validationError
-          ? validationError.message
-          : 'The required fields can not be empty'
-      });
-    } else {
+        this._saveConnection(currentConnection);
+      } catch (error) { /* Ignore saving invalid connection string */ }
```

Comment:
I think we should be handling errors that might occur when saving connections, however it's saving the connection in storage mixin in a pretty unhandleable way at the moment so I don' think there's much we can do right now.